lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,510 @@
+                            }).join('');
+          result.payload = tostr;
+          return tostr;
+        }
+    };
+  };
+
+  var buildAckPacket = function() {
+    return buildPacket(ASCII_CONTROL.ACK, 1);
+  };
+
+  var buildHeaderPacket = function() {
+    var cmd = 0x3a;  // just anything 'x'
+    return buildPacket(cmd, 1);
+  };
+
+  var extractPacket = function (bytes) {
+    var packet = {
+      bytes: bytes,
+      valid: false,
+      packet_len: 0,
+      payload: null
+    };
+
+    var plen = bytes.length;
+    var packet_len = struct.extractByte(bytes, 0);
+
+    if (packet_len > plen) {
+      return packet;
+    }
+
+    //discard the length byte from the begining
+    var tmpbuff = new ArrayBuffer(packet_len);
+    struct.copyBytes(tmpbuff, 0, bytes, packet_len, 1);
+    packet.bytes = tmpbuff;
+
+    // we now have enough length for a complete packet, so calc the CRC
+    packet.packet_len = packet_len;
+    packet.valid = true;
+
+    return packet;
+  };
+
+
+  var buildHeaderCmd = function() {
+    return {
+      packet: buildAckPacket(),
+      parser: function (result) {
+        var tostr = _.map(result.bytes,
+                          function(e){
+                            return String.fromCharCode(e);
+                          }).join('');
+        result.payload = tostr;
+        return tostr;
+      },
+    };
+  };
+
+  var parseHeader = function (s){
+    var data = s.split('\n').filter(function(e){ return e !== '';});
+    var header = data.shift();
+    if(verifyChecksum(header)){
+      var patient = data.shift();
+      var lineFeed = data.pop();
+      var pString = header.split('|');
+      var pInfo = pString[4].split('^');
+      var sNum = pInfo[2].split('-');
+      var records = data;
+
+      var devInfo = {
+        model: pInfo[0],
+        serialNumber: sNum[1],
+        nrecs: pString[6],
+        rawrecords: records
+      };
+
+      return devInfo;
+    }else{
+      return null;
+    }
+  };
+
+  function verifyChecksum(record){
+      var str = record.trim();
+      var data = str.split(String.fromCharCode(ETB));
+      var check = data[1];
+      var sum = 0;
+      var n = record.slice(0, record.length - 3);
+
+      _.map(n, function(e){
+          if(e.charCodeAt(0) !== STX){
+              sum += e.charCodeAt(0);
+          }
+      });
+
+      if((sum % 256) !== parseInt(check, 16)){
+          return null;
+      }else{
+          return data[0];
+      }
+  }
+
+  var parseDataRecord = function (str, callback){
+    var data = verifyChecksum(str);
+    if(data){
+      var result = data.trim().match(/^.*\d+R\|(\d+).*Glucose\|(\d+)\|(\w+\/\w+).*\|{2}(.*)\|{2}(\d{12})$/).slice(1,6);
+      callback(null, result);
+    }else{
+      throw( new Error('Invalid record data') );
+    }
+
+  };
+
+  var getAnnotations = function (annotation){
+    if (annotation == '>') {
+      return [{
+        code: 'bg/out-of-range',
+        value: 'high',
+        threshold: 600
```

Comment:
According to my read of the spec, you should be able to get the threshold out of the header record as glucose HI / LO range rather than hard-coding as 600. Can you do that? Again to guard against devices that could be in mmol/L...